pageserver/src/tenant.rs

```diff
@@ -1719,18 +1809,23 @@ impl Tenant {
     /// This function waits for the tenant to become active if it isn't already, before transitioning it into Stopping state.
     ///
     /// This function is not cancel-safe!
-    async fn set_stopping(&self, progress: completion::Barrier) -> Result<(), SetStoppingError> {
+    async fn set_stopping(
```

Comment:
Could you include in which cases we (do not)? want to allow the transition from loading? I haven't yet looked at the callers.

Reply:
In case of resumed deletion we need allow the transition from Loading to Stopping. Previously it was not allowed. So set_stopping was waiting for loading to complete and thus blocked forever

Reply:
Aah good point!

Reply:
```suggestion
    ///
    /// `allow_transition_from_loading` is needed for the special case of loading task deleting the tenant.
    async fn set_stopping(
```